tutorials/tutorial-howto-simulate-iot-sensor-data.md

```diff
@@ -0,0 +1,304 @@
+basic IoT sensor dataset on PostgreSQL, or TimescaleDB.
+
+*Note: For creating a more advanced simulated IoT dataset, please
+try the Time-series Benchmarking Suite (TSBS) ([Github][github-tsbs]).*
+
+### Prerequisites [](prereqs)
+
+* A PostgreSQL or TimescaleDB
+([installation instructions][get-timescale-cloud]) instance
+
+* `psql` ([installation instruction][get-psql])
+
+>:TIP: If you install PostgreSQL or TimescaleDB locally, `psql` may come
+> preinstalled
+
+### Step 1 - Set up your tables [](step1)
+First, connect to your database via `psql`.
+
+Second, create the “sensors” and “sensor_data” tables:
+
+```sql
+CREATE TABLE sensors(
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(50),
+  location VARCHAR(50)
+);
+```
+
+```sql
+CREATE TABLE sensor_data (
+  time TIMESTAMPTZ NOT NULL,
+  sensor_id INTEGER,
+  temperature DOUBLE PRECISION,
+  cpu DOUBLE PRECISION,
+  FOREIGN KEY (sensor_id) REFERENCES sensors (id)
+);
+```
+
+Third (only if you are using TimescaleDB), convert the `sensor_data`
+table into a [`hypertable`][docs-hypertable]:
+
+```sql
+SELECT create_hypertable('sensor_data', 'time');
+```
+
+Fourth, populate the sensors table with 4 sensors:
+
+```sql
+INSERT INTO sensors (type, location) VALUES
+('a','floor'),
+('a', 'ceiling'),
+('b','floor'),
+('b', 'ceiling');
+```
+
+Fifth, verify that the sensors were created correctly:
+
+```sql
+SELECT * FROM sensors;
+```
+
+After running that last SQL statement, you should see something like
+this:
+
+```
+ id | type | location 
+----+------+----------
+  1 | a    | floor
+  2 | a    | ceiling
+  3 | b    | floor
+  4 | b    | ceiling
+(4 rows)
+```
+
+### Step 2 - Create the simulated IoT sensor data [](step2)
+
+*Note: for the following sections we’ll share the results of our queries as an example, but since the tutorials generates random data every time it is run, your results will look different (but will be structured the same way).*
+
+First, generate a dataset of simulated data for one sensor, recording data every 5 minutes for the past 24 hours:
+
+```sql
+SELECT 
+  generate_series(now() - interval '24 hour', now(), interval '5 minute') AS time,
+  random() AS cpu,
+  random()*100 AS temperature;
+```
+
+SAMPLE OUTPUT (*Note: Your data will look different but should have the same structure*):
+
+```
+             time              |         cpu          |    temperature    
+-------------------------------+----------------------+-------------------
+ 2020-03-31 15:55:31.744218+00 |   0.0306301130913198 |  75.7227655500174
+ 2020-03-31 16:00:31.744218+00 |    0.626729523297399 |  20.1422684360296
+ 2020-03-31 16:05:31.744218+00 |    0.526347786653787 |  87.2947076335549
+ 2020-03-31 16:10:31.744218+00 |    0.169700589030981 |  21.1996510624886
+ 2020-03-31 16:15:31.744218+00 |     0.71232553711161 |  91.3718110416085
+ ```
+
+Second, generate a similar dataset for each of our four sensors and insert into our sensor_data table:
+
+```sql
+!-- For sensor 1
+WITH simulated_data 
```

Comment:
why are you using CTEs here? could also generate the data using a single query instead of repeating the same query 4 times.

```
INSERT INTO sensor_data (time, sensor_id, cpu, temperature)
SELECT
  time,
  sensor_id,
  random() AS cpu,
  random()*100 AS temperature
FROM generate_series(now() - interval '24 hour', now(), interval '5 minute') AS g1(time), generate_series(1,4,1) AS g2(sensor_id)
```

Reply:
Ah this is great. Ty! Will do.